uid/snowid.go

```diff
@@ -0,0 +1,192 @@
+package uid
```

Comment:
Ya, good question. Everything here should be from the upstream repo. The changes I made (that I haven't noted yet) are:

* unexported `epoch`, `nodeBits`, `stepBits`, so they can't be modified outside this package
* `epoch` is set to the value we were using (instead of the twitter value which was previously the default)
* `Bytes` and `String` now returned the base58 encoded ID. In the upstream they would return the int64 value, and our wrapper defined a different `String`. So now the `String` matches what our wrapper did (return the base58 encoded value).
* method receiver was renamed from `f` to `id`
* `MarshalJSON` and `UnmarshalJSON` were replaced by `MarshalText` and `UnmarshalText` (which we had in our wrapper). The JSON decode/encoder knows to call those, and removes the need for us to handle the `""`.

That should be it. Everything else is noted below, or was a change to the tests. There were also lots of things deleted, but we don't use those things.